src/gossip/gossip-startup.lisp

```diff
@@ -96,31 +96,33 @@ with this list if desired."
 (let ((gossip-inited nil))
   (defun gossip-init (&optional (cmd))
     "Call this once before using gossip system. Should return true if everything succeeded."
-    ; make actor:pr calls go to global log.
+
     ; NOTE: We're not just substituting *logging-actor* here because we want the timestamp to
     ;   be made ASAP after function is called, not when logging finally happens.
     (case cmd
       (:init
        #+IGNORE ; #+OPENMCL ; these hemlock streams are just too slow when they get to a couple thousand lines
        ; in CCL, we'll just inspect the *log*
        (if (find-package :gui)
-           (setf *logstream* (funcall (intern "MAKE-LOG-WINDOW" :gui) "Emotiq Log"))
-           (setf *logstream* *standard-output*))
+           (setf emotiq:*notestream* (funcall (intern "MAKE-LOG-WINDOW" :gui) "Emotiq Log"))
+           (setf emotiq:*notestream* *error-output*))
        #+(or (not :openmcl) (not :shannon))
```

Comment:
Please write a comment as to what you intend to do with the logging ouput from `emotiq/logging`, as the intentions are unclear from the code as written, and I don't think this does what you want.

As I understand this conditionals, if the feature :shannon is not present you unconditionally muffle emotiq:note for all users if not running on CCL. If the feature :shannon is present on a Lisp other than CCL, you set effectively have

(setf emotiq:notestream error-output nil)

which maybe fails to compile, and certainly might behave non-standardly.

Part of clarifying things would be to remove :shannon to reflect semantics other than the current developer is not Shannon. Doesn't it mean :running-ccl-hemlock or :using-hemlock?

Reply:
Mark, I didn't add the conditional for :SHANNON; you did. So I'm confused as to why you're confused.
https://github.com/emotiq/emotiq/commit/30bc299aa2888b0f79feb9f04b1effcf6e466dcf

Furthermore, the line `(setf emotiq:notestream error-output nil)` does not exist, and the code compiles just fine.

There are two points I'm trying to address herein:
1. Specifying `*error-output*` as the output stream for note messages is the wrong thing. Notes are not errors. Note output needs to go to its own stream whose connotation is "note" and which can be rebound as needed.
2. In CCL using the IDE on a Mac, every output written to `*standard-output*` or `*error-output*` gets sent to a listener associated with the current thread. In multithreaded code, this results in a complete mess on the screen with a separate listener window popping up whenever a new thread prints a note message. Therefore, in CCL using the IDE on a Mac, it's *very* important that the output of `note` not go to `t` or `*standard-output*` or `*error-output*`. In my case, I prefer it to be bound to nil, so I don't see any note output whatsoever. I have my own logging system in gossip which sends all messages to `gossip:*log*`, which is a vector of lisp objects whose events have accurate timestamps and are guaranteed to be properly serialized.

Reply:
I have not said that the code does not compile.

I have asked you to document the meaning of the features in the code

Reply:
@svspire 

I would then rename the `:shannon` feature to `:ccl-with-ide`.

Reply:
All fixed now.